Analysis/Producer.py

```diff
@@ -63,7 +69,22 @@ def defineWeights(df_dict):
         weight_str = " * ".join(weight_names)
         df_dict[sample]=df_dict[sample].Define("weight",weight_str)
 
-def createHistograms(df_dict, var):
+
+def Estimate_QCD(var, df_dict, hist_dict, model, deepTauVersion):
+    df_data = df_dict['data']
+    data_2 = df_data.Filter('region_B').Sum("weight").GetValue()
+    hist_data_3 = df_data.Filter('region_C').Histo1D(model, var, "weight").Clone()
+    data_4 = df_data.Filter('region_D').Sum("weight").GetValue()
+    kappa = data_2/data_4
+    for sample in df_dict.keys():
+        if sample=='data' or sample in signals:
+            continue
+        hist_3 = df_dict[sample].Filter("region_C").Histo1D(model,var,"weight").Clone() # OS non-iso
+        hist_data_3.Add(hist_3, -1)
+    hist_data_3.Scale(kappa)
+    return hist_data_3
```

Comment:
only store the ingredients. do the actual computation in the final loop where `GetValue` is called for all histograms